src/createStaticPackage.js

```diff
@@ -21,15 +23,56 @@ const IFRAME_CONTENT = `
 </html>
 `;
 
+/**
+ * Gets all files in a directory and all of its subdirectories
+ *
+ * The returned value is sorted for deterministic output.
+ *
+ * @param {string} dir
+ * @returns {Array<string>}
+ */
+function getFilesRecursive(dir) {
+  const files = [];
+  const dirs = [dir];
+
+  while (dirs.length > 0) {
+    const currentDir = dirs.shift();
+    // Node 18 adds `recursive` option to `fs.readdirSync`, which would
+    // make this function simpler.
+    const dirents = fs.readdirSync(currentDir, { withFileTypes: true });
+
+    for (const dirent of dirents) {
+      const res = path.resolve(currentDir, dirent.name);
+
+      if (dirent.isDirectory()) {
+        // This is a directory, so we are going to add it to the list of directories to recurse into.
+        dirs.push(res);
+        files.push(res);
```

Comment:
I originally didn't include this line, but we have an integration test that asserts on the full contents of the archive, and the directories were previously present as distinct entries. Adding this line fixed that test. I'm not sure how important this is though, but I wanted to preserve existing functionality to be safe.